graphql_compiler/global_utils.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+from collections import OrderedDict
+from itertools import chain
+
+import six
+
+
+def _get_dag_with_keys_in_reverse_topological_order(dag):
+    """Return an OrderedDict representing a DAG with keys ordered in reversed topological order.
+
+    Args:
+        dag: dict, string -> set of strings, mapping each vertex in the directed acyclic graph to
+             the set of vertices it has a directed edge to.
+
+    Returns:
+        OrderedDict, str -> str, representing a directed acyclic graph (DAG) with keys ordered in
+        reversed topological order.
+    """
+    def traverse_vertex_subgraph(vertex_name, processed_vertices, current_trace):
+        """Traverse a vertex and its edges recursively in a depth-first-search manner.
+
+        Args:
+            vertex_name: string, the name of the vertex whose subgraph is traversed.
+            processed_vertices: set of strings, a set of vertices that have already been visited.
+                                May be mutated by this function.
+            current_trace: list of strings, list of vertices visited during the traversal.
+                           May be mutated by this function.
+
+        Returns:
+            list of strings, representing the set of vertices that have been first visited in this
+            traversal ordered in the reverse order they were visited.
+        """
+        if vertex_name in processed_vertices:
+            return []
+
+        if vertex_name in current_trace:
+            raise AssertionError(
+                'Encountered self-reference in dependency chain of {}'.format(vertex_name))
+
+        vertex_list = []
```

Comment:
A more descriptive name could be helpful, perhaps something like vertices_in_reverse_traversal_order (maybe that's too long)?